extensions/amp-video/0.1/amp-video.js

```diff
@@ -454,11 +462,30 @@ class AmpVideo extends AMP.BaseElement {
       sources.unshift(srcSource);
     }
 
-    // Only cached sources are added during prerender.
+    // Only cached sources are added during prerender, with all qualities supported.
     // Origin sources will only be added when document becomes visible.
     sources.forEach((source) => {
       if (this.isCachedByCDN_(source)) {
-        this.video_.appendChild(source);
+        source.remove();
+        const qualities = Object.keys(AMP_QUALITY_BITRATES);
+        const origType = source.getAttribute('type');
+        const origSrc = source.getAttribute('amp-orig-src');
+        qualities.forEach((quality) => {
+          const cachedSource = addParamsToUrl(source.src, {
+            'amp_quality': quality,
+          });
+          const currSource = this.createSourceElement_(
+            cachedSource,
+            origType,
+            AMP_QUALITY_BITRATES[quality]
+          );
+          if (qualities.indexOf(quality) == qualities.length - 1) {
```

Comment:
- Please use the `index` param of the forEach callback
- Use strict equality operator